lib/ansible/modules/notification/mattermost.py

```diff
@@ -0,0 +1,160 @@
+short_description: Send Mattermost notifications
+description:
+    - The M(mattermost) module sends notifications to U(http://your.mattermost.url) via the Incoming WebHook integration
+version_added: "2.3"
+author: "Benjamin Jolivot (@bjolivot)"
+options:
+  url:
+    description:
+      - Mattermost url (i.e. http://mattermost.yourcompany.com)
+    required: true
+  api_key:
+    description:
+      - Mattermost webhook api key. Log into your mattermost site, go to
+        Menu -> Integration -> Incomming Webhook -> Add Incomming Webhook
+        This will give you full URL. api_key is the last part.
+        http://mattermost.example.com/hooks/C(API_KEY)
+    required: true
+  text:
+    description:
+      - Text to send. Note that the module does not handle escaping characters.
+    required: true
+  channel:
+    description:
+      - Channel to send the message to. If absent, the message goes to the channel selected for the I(api_key).
+    required: false
+    default: None
+  username:
+    description:
+      - This is the sender of the message (Username Override need to be enabled by mattermost admin, see mattermost doc, default C(Ansible)).
+    required: false
+    default: "Ansible"
+  icon_url:
+    description:
+      - Url for the message sender's icon (default C(https://www.ansible.com/favicon.ico))
+    required: false
+  validate_certs:
+    description:
+      - If C(no), SSL certificates will not be validated. This should only be used
+        on personally controlled sites using self-signed certificates.
+    required: false
+    default: 'yes'
+    choices:
+      - 'yes'
+      - 'no'
+"""
+
+EXAMPLES = """
+- name: Send notification message via Mattermost
+  mattermost:
+    url: http://mattermost.example.com
+    api_key: my_api_key
+    text: '{{ inventory_hostname }} completed'
+
+- name: Send notification message via Slack all options
+  mattermost:
+    url: http://mattermost.example.com
+    api_key: my_api_key
+    text: '{{ inventory_hostname }} completed'
+    channel: notifications
+    username: 'Ansible on {{ inventory_hostname }}'
+    icon_url: http://www.example.com/some-image-file.png
+"""
+
+RETURN = '''
+payload:
+    description: Mattermost payload
+    returned: success
+    type: string
+webhook_url:
+    description: URL the webhook is sent to
+    returned: success
+    type: string
+'''
+
+
+def main():
+    module = AnsibleModule(
+        supports_check_mode=True,
+        argument_spec = dict(
+            url         = dict(type='str', required=True),
+            api_key     = dict(type='str', required=True,  no_log=True),
+            text        = dict(type='str', required=True),
+            channel     = dict(type='str', required=False,  default=None),
+            username    = dict(type='str', default='Ansible'),
+            icon_url    = dict(type='str', default='https://www.ansible.com/favicon.ico'),
+            validate_certs = dict(default='yes', type='bool'),
+        )
+    )
```

Comment:
Can we add check-mode support ?

Check-mode support in this case would probably do as much as possible without actually sending out the actual message. If there is a way to test sending as well (like some sort of /dev/null bucket) that would be great. But if not, if the module would test credentials and connection in check-mode, that in itself would be sufficient.